src/main/java/org/folio/rest/impl/PatronNoticePoliciesAPI.java

```diff
@@ -159,38 +188,84 @@ public void putPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicy
                   .respond422WithApplicationJson(createNotUniqueNameErrors(entity.getName()))));
               return;
             }
-            asyncResultHandler.handle(Future.succeededFuture(
+            asyncResultHandler.handle(succeededFuture(
               PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse
                 .respond500WithTextPlain(update.cause())));
             return;
           }
 
           if (update.result().getUpdated() == 0) {
-            asyncResultHandler.handle(Future.succeededFuture(
+            asyncResultHandler.handle(succeededFuture(
               PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse
                 .respond404WithTextPlain(NOT_FOUND)));
             return;
           }
-          asyncResultHandler.handle(Future.succeededFuture(
+          asyncResultHandler.handle(succeededFuture(
             PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse.respond204()));
         });
       } catch (Exception e) {
         logger.error(e);
-        asyncResultHandler.handle(Future.succeededFuture(
+        asyncResultHandler.handle(succeededFuture(
           PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse.respond500WithTextPlain(e)));
       }
     });
   }
 
   private Errors createNotUniqueNameErrors(String name) {
     Error error = new Error();
-    error.setMessage(String.format("'%s' name in not unique", name));
+    error.setMessage(format("'%s' name in not unique", name));
     error.setCode(STATUS_CODE_DUPLICATE_NAME);
-    return new Errors().withErrors(Collections.singletonList(error));
+    return new Errors().withErrors(singletonList(error));
+  }
+
+  private Future<CirculationRules> findCirculationRules(PostgresClient pgClient) {
+
+    Future<Results<CirculationRules>> future = Future.future();
+    pgClient.get(CIRCULATION_RULES_TABLE, CirculationRules.class, new Criterion(), false, false, future.completer());
+    return future.map(Results::getResults)
+      .compose(list -> list.size() == 1 ? succeededFuture(list) :
+        failedFuture(new IllegalStateException("Number of records in circulation_rules table is " + list.size())))
+      .map(list -> list.get(0));
+  }
+
+  private Future<Void> deleteNoticePolicyById(PostgresClient pgClient, String id) {
+
+    Criteria criteria = new Criteria()
+      .addField("'id'")
+      .setOperation(OP_EQUAL)
+      .setValue("'" + id + "'");
+
+    Future<UpdateResult> future = Future.future();
+    pgClient.delete(PATRON_NOTICE_POLICY_TABLE, new Criterion(criteria), future.completer());
+
+    return future.map(UpdateResult::getUpdated)
+      .compose(updated -> updated > 0 ? succeededFuture() : failedFuture(new NotFoundException(NOT_FOUND)));
+  }
+
+  private Response mapExceptionToResponse(Throwable t) {
+
+    if (t.getClass() == NotFoundException.class) {
```

Comment:
Is there a reason why this is does a reference comparison, rather than using `instanceOf`?

Reply:
I'm looking for matching an exact class therefore I don't see any reason to use `instanceOf` here.

Reply:
Ok (I don't follow why it's necessary to match the exact class, but that's ok).

This code will break if the exception is null, is it worth being defensive at the beginning of the method?

Reply:
If the exception is null, the method returns 500 response. Probably it is not a proper behavior but exception cannot be null here as the method is invoked only when the future is failed. So I think it does not make a lot of sense to handle null exception here.

Reply:
Hmm, if the exception is null, I believe there is likely to be no response at all, because `t.getClass()` will throw a fresh `NullPointerException`.

I'm ok with an unknown cause resulting in a 500. I can't think of an alternative, given the information available?

Does a failed future guarantee that there is an exception present?

Reply:
Sorry, you're right about `NullPointerException`. 

Yes, as I far as I know a failed future guarantee that there is an exception present (if we fail a future without any exception, the future will be failed with `io.vertx.core.impl.NoStackTraceThrowable`).

Reply:
Currently `mapExceptionToResponse` method is used only to map exceptions for one endpoint. The method will be extending to handle more cases if we will want to use it for other endpoints.

Reply:
Ok, if we are confident this won't be invoked with null, then we can resolve this